src/main/java/sirius/db/qdrant/Search.java

```diff
@@ -90,16 +92,17 @@ public <T> List<T> queryPayload(Class<T> fieldType, String field, int maxResults
         }
 
         JSONObject response = qdrantDatabase.execute(QdrantDatabase.Method.POST,
-                                                     QdrantDatabase.URI_PREFIX_COLLECTIONS + collection + "/points/search",
+                                                     QdrantDatabase.URI_PREFIX_COLLECTIONS
+                                                     + collection
+                                                     + "/points/search",
                                                      query);
         JSONArray points = response.getJSONArray("result");
-
-        List<T> payloadData = new ArrayList<>(points.size());
-        for (JSONObject point : points.toJavaList(JSONObject.class)) {
-            payloadData.add(fieldType.cast(point.getJSONObject("payload").get(field)));
-        }
-
-        return payloadData;
+        return points.stream()
+                     .map(JSONObject.class::cast)
+                     .map(point -> new Match(point.getString("id"),
```

Comment:
**Optional:** A constructor that takes a `JSONObject` could be added to internalize parsing/fetching the correct fields